x-pack/plugins/stack_connectors/public/connector_types/slack_api/slack_connectors.tsx

```diff
@@ -33,15 +42,123 @@ const getSecretsFormSchema = (docLinks: DocLinksStart): SecretsFieldSchema[] =>
   },
 ];
 
+const getConfigFormSchemaAfterSecrets = (
+  options: EuiComboBoxOptionOption[],
+  isLoading: boolean,
+  isDisabled: boolean
+): ConfigFieldSchema[] => [
+  {
+    id: 'allowedChannels',
+    isRequired: false,
+    label: i18n.ALLOWED_CHANNELS,
+    helpText: (
+      <FormattedMessage
+        id="xpack.stackConnectors.components.slack_api.allowedChannelsText"
+        defaultMessage="By default, the connector can access all channels within the scope of the Slack app."
+      />
+    ),
+    type: 'COMBO_BOX',
+    euiFieldProps: {
+      isDisabled,
+      isLoading,
+      noSuggestions: false,
+      options,
+    },
+  },
+];
+
 const SlackActionFields: React.FC<ActionConnectorFieldsProps> = ({ readOnly, isEdit }) => {
-  const { docLinks } = useKibana().services;
+  const {
+    docLinks,
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const form = useFormContext();
+  const { setFieldValue } = form;
+  const [formData] = useFormData({ form });
+  const [authToken, setAuthToken] = useState('');
+  const [channels, setChannels] = useState<EuiComboBoxOptionOption[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const abortCtrlRef = useRef(new AbortController());
+  const isMounted = useRef(false);
+
+  async function fetchChannels(newAuthToken: string) {
+    setIsLoading(true);
+    setChannels([]);
+    setFieldValue('config.allowedChannels', []);
+    isMounted.current = true;
+    abortCtrlRef.current.abort();
+    abortCtrlRef.current = new AbortController();
+
+    try {
+      const res = await http.post<GetChannelsResponse>(
```

Comment:
Not a fan of having the fetching logic inside the component.

Also I was talking to Christos abort signals and we both really liked how react query handles it, It supports abort signal and cancellation by query key (instead of passing around an abort signal ref https://tanstack.com/query/v4/docs/react/guides/query-cancellation#manual-cancellation).

Request cancellation becomes much less of an issue with react query in the first place since every request is named spaced by query key and cached. Even if the component mounts, the result is cached and properly GC'd when not used, no more memory leaks.

Reply:
I agree with @JiaweiWu. Given that there is no time for that (I assume this PR is for 8.9), I think we can export the logic to each own hook and its own file.

Reply:
I will change it to react query, not a big deal!